### ISSUE TITLE
Improve Github release documentation

### ISSUE THREAD

**Issue by tbcrawford** [kotest/kotest#3129]
## Request
As it stands now, from my perspective, the change log for Kotest appears to be a bit hidden. In order to get to the change log from the website, you have to scroll to the bottom of the page and look at the footer to get to the change log. Getting to the change log from the README seems modestly sufficient for now. However, with that said, I think adding the changes for each release to the release notes in Github would provide another method for change discoverability. The Spring team does a great job of this with their [release notes](https://github.com/spring-projects/spring-boot/releases). 

## Possible Solutions
1. Use [release drafter](https://github.com/release-drafter/release-drafter) to automate release note generation
2. Copy change log documentation for each release that normally goes into [changelog.md](https://github.com/kotest/kotest/blob/master/documentation/docs/changelog.md) to the release notes markdown section for each release.

## Benefits
I believe that adding documentation to each release in Github will provide greater change discoverability and tools such as [renovate](https://github.com/renovatebot/renovate) may be able to pick the changes up and show them in automatically created pull requests for developers to review so they don't have to find the change log for each release.

**Comment by ahatzz11** [comment]
Another possible solution here is to use the GitHub auto-generated changelogs in the GitHub Releases. This will automatically generate release notes based on the PRs that have been merged since the last release. A `.github/release.yml` file is allowed to customize the sections of each release as well.

<img width="303" alt="image" src="https://user-images.githubusercontent.com/6256032/181568235-584739e5-b0fb-4c95-a8ca-f3b632d662d7.png">

I'm not quite sure how the github releases are created today - I don't see anything automated in the `.github/workflows`. If this is a manual step right now, a quick payoff would probably just be clicking the generate release notes button (as shown above). If there is an automated thing creating the GitHub releases, the GitHub API does support the creation of auto generated release notes.

**Comment by sksamuel** [comment]
I think we tried release drafter in the past but it doesn't include stuff pushed straight to master (which is another topic I guess :) ). And lots of PRs are cruft in my opinion. No one looking at a renovate PR cares that we fixed a typo in docs. So maybe your second option is a good solution - copying the changelog notes into the release section in github?

@ahatzz11 Right now we create the changelogs manually based on a git log diff.
